pkgs/development/tools/build-managers/build2/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, lib, fetchurl, sqlite }:
+
+let
+  version = "0.13.0";
+in
+stdenv.mkDerivation {
+  pname = "build2";
+  inherit version;
+
+  src = fetchurl {
```

Comment:
This tarball is an amalgamation which vendors the dependencies (libbutl, libpkgconf) and bootstraps the build2 toolchain. IMO this should only be used as a bootstrap derivation for the final build2 and enforced with disallowedReferences.

Also, it looks like this is installing bdep and bpkg to the same prefix - as they are standalone tools, they should probably be in separate derivations. Nixpkgs will only need to use build2 internally, so this will reduce build closure size.

Reply:
b bdep and bpkg arent much use standalone as far as i can tell
I do not have enough know-how to correctly understand you first point.

Reply:
> This tarball is an amalgamation which vendors the dependencies (libbutl, libpkgconf) and bootstraps the build2 toolchain. IMO this should only be used as a bootstrap derivation for the final build2 and enforced with disallowedReferences.

This would require to build the toolchain once with the vendored deps and then a second time purely in nixpkgs with https://github.com/NixOS/nixpkgs/blob/5da845a7e06ad35a181c532c1469e00df60288f6/pkgs/applications/editors/neovim/default.nix#L95 set.

This makes sure that when the final build contains a reference to the bootstratp tools the build fails.

Reply:
For the bootstrap process see https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml#install
Here we skip stage 4/5

Reply:
We'll want to install each tool (b, bpkg, bdep) and library (libbutl, libbpkg, libodb, libodb-sqlite) to a different store path though, the monolithic bootstrap puts everything in the same prefix. It's probably not a big deal for now since no other packages depend on these yet, as long as you're all right with a big refactor in the near future.

Reply:
Here's my current WIP patch so you can see what I mean: https://github.com/r-burns/nixpkgs/commit/62ca29f96c3e5135c5dcba9ab222498683ea275a

Reply:
i basically packaged build2 this way to scratch my own itch. I'm ok if this pr is not merged or radically altered in the future.

Reply:
but i see your point now :]